File: minipress_app/lib/models/Article.dart
class Article {
  final int? id;
  final String? title;
  final DateTime? dateCreation;
  final String? auteur;
  final String? resume;
  final String? contenu;
  final String? image;
  final int? idCateg;
  final String? idAuteur;

  Article(
      {this.id,
      this.title,
      this.dateCreation,
      this.auteur,
      this.resume,
      this.contenu,
      this.image,
      this.idCateg,
      this.idAuteur});

  factory Article.fromJson(Map<String, dynamic> json) {
    return Article(
      id: json['id'],
      title: json['titre'],
      dateCreation: DateTime.parse(json['date_creation']),
      auteur: json['auteur'],
      resume: json['resume'],
      contenu: json['contenu'],
      image: json['image'],
      idCateg: json['cat_id'],
      idAuteur: json['id_auteur'],
    );
  }
}

File: minipress.core/press.api/src/models/Article.php
<?php

namespace press\api\models;

use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;

class Article extends Model
{
    public $incrementing = false;
    public $keyType = 'string';
    public $timestamps = false;
    public $fillable = ['id', 'titre', 'date_creation', 'auteur', 'resume', 'contenu', 'date_publication', 'image', 'idCateg'];
    protected $table = 'article';
    protected $primaryKey = 'id';

    function categories(): BelongsTo
    {
        return $this->belongsTo(Categorie::class, 'idCateg');
    }
}